src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -61,29 +92,149 @@ CHLSLCompiler::~CHLSLCompiler()
+                    system::ILogger::ELL_ERROR);
+                return;
+            }
+
+            // Stage
+            stageStr = std::wstring(s.begin(), underscorePositions[0]);
+            if (shaderStageOverrideFromPragma != IShader::ESS_UNKNOWN) // replace first 2 characters if shaderStageOverrideFromPragma != Unknown
+            {
+                if (!overrideStageStr) {
+                    logger.log("Invalid shader stage with int value '%i'.\nThis value does not have a known string representation.",
+                        system::ILogger::ELL_ERROR, shaderStageOverrideFromPragma);
+                    return;
+                }
+                stageStr = std::wstring(overrideStageStr);
+            }
+
+            // Version
+            int size = underscorePositions.size();
+            auto secondLastUnderscore = underscorePositions[size - 2];
+            auto lastUnderscore = underscorePositions[size - 1];
+            majorVersionString = std::wstring(secondLastUnderscore + 1, lastUnderscore);
+            minorVersionString = std::wstring(lastUnderscore + 1, s.end());
+            try
+            {
+                int major = std::stoi(majorVersionString);
+                int minor = std::stoi(minorVersionString);
+                if (major < MajorReqVersion || (major == MajorReqVersion && minor < MinorReqVersion)) 
+                {
+                    // Overwrite the version 
+                    logger.log("Upgrading shader stage version number to %i %i", system::ILogger::ELL_DEBUG, MajorReqVersion, MinorReqVersion);
+                    arguments[foundShaderStageArgumentValueIdx] = stageStr + L"_" + std::to_wstring(MajorReqVersion) + L"_" + std::to_wstring(MinorReqVersion);
+                }
+                else 
+                {
+                    // keep the version as it was
+                    arguments[foundShaderStageArgumentValueIdx] = stageStr + L"_" + majorVersionString + L"_" + minorVersionString;
+                }
+                return;
+            }
+            catch (const std::invalid_argument& e) {
+                logger.log("Parsing shader version failed, invalid argument exception: %s", system::ILogger::ELL_ERROR, e.what());
+
+            }
+            catch (const std::out_of_range& e) {
+                logger.log("Parsing shader version failed, out of range exception: %s", system::ILogger::ELL_ERROR, e.what());
+            }
+            
+            // In case of an exception 
+            arguments[foundShaderStageArgumentValueIdx] = stageStr + L"_" + std::to_wstring(MajorReqVersion) + L"_" + std::to_wstring(MinorReqVersion);
+            return;
+        }
+        else {
+            logger.log("invalid shader stage '%s' argument, expecting a string of length >= 6 ", system::ILogger::ELL_ERROR, s);
+        }
+	}
```

Comment:
First thing: fix the formatiing/indentation !

Second thign: start using this pattern
```cpp
if (cond0)
{
   if (cond1)
   {
      success Case
   }
}

Failure Case
```
or this one
```cpp
Set Failure Case
if (cond0)
{
   if (cond1)
   {
      Overwrite Failure with Success Case
   }
}
```

Right now this code is unreadable mess, and if you hit the else case of `s.length()<6` you'll only end up with an error being logged, not a reasonable fallback.